### ISSUE TITLE
Ubuntu 22.04, k3s, cilium v1.12.2: error dialing backend: EOF

### ISSUE THREAD

**Issue by sanderterschure** [cilium/cilium#21405]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Over the past couple of days I try to figure out how to get cilium CNI to work on k3s running on Ubuntu server 22.04. It started from having multiple nodes in my k3s cluster to scoping it down to a single node cluster to get it to work. 

My Ubuntu node is running as a virtual machine on my Supermicro server running Vmware ESXI version 7. It is configured with 2 vCPU and 4GB RAM and a SSD disk of 80GB. 

When installing and configuring Ubuntu, k3s and cilium it looks like it is working. But eventually when running `cilium connectivity test` randomly tests will fail. Each test run has different tests that are failing. 

To reproduce my setup I use the following commands:

1. Fresh install of Ubuntu server 22.04

[Ubuntu 22.04](https://ubuntu.com/download/server)

2. Install k3s

```
curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC='--flannel-backend=none --disable-network-policy --disable=traefik --disable=metrics-server' sh -
```

3. Set KUBECONFIG file

```
export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
```

4. Install `cilium-cli`

```
CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
CLI_ARCH=amd64
if [ "$(uname -m)" = "aarch64" ]; then CLI_ARCH=arm64; fi
curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
sha256sum --check cilium-linux-${CLI_ARCH}.tar.gz.sha256sum
sudo tar xzvfC cilium-linux-${CLI_ARCH}.tar.gz /usr/local/bin
rm cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
```

6. Install cilium CNI

```
root@test:/usr/local/bin# cilium install
🔮 Auto-detected Kubernetes kind: K3s
ℹ️  Using Cilium version 1.12.2
🔮 Auto-detected cluster name: default
🔮 Auto-detected datapath mode: tunnel
🔮 Auto-detected kube-proxy has not been installed
ℹ️ Cilium will fully replace all functionalities of kube-proxy
ℹ️  helm template --namespace kube-system cilium cilium/cilium --version 1.12.2 --set cluster.id=0,cluster.name=default,encryption.nodeEncryption=false,k8sServiceHost=127.0.0.1,k8sServicePort=6443,kubeProxyReplacement=strict,operator.replicas=1,serviceAccounts.cilium.name=cilium,serviceAccounts.operator.name=cilium-operator,tunnel=vxlan
ℹ️  Storing helm values file in kube-system/cilium-cli-helm-values Secret
🔑 Created CA in secret cilium-ca
🔑 Generating certificates for Hubble...
🚀 Creating Service accounts...
🚀 Creating Cluster roles...
🚀 Creating ConfigMap for Cilium version 1.12.2...
🚀 Creating Agent DaemonSet...
🚀 Creating Operator Deployment...
⌛ Waiting for Cilium to be installed and ready...
✅ Cilium was successfully installed! Run 'cilium status' to view installation health
```

7. Check cilium status after install

```
root@test:/home/sander# cilium status
    /¯¯\
 /¯¯\__/¯¯\    Cilium:         OK
 \__/¯¯\__/    Operator:       OK
 /¯¯\__/¯¯\    Hubble:         disabled
 \__/¯¯\__/    ClusterMesh:    disabled
    \__/

DaemonSet         cilium             Desired: 1, Ready: 1/1, Available: 1/1
Deployment        cilium-operator    Desired: 1, Ready: 1/1, Available: 1/1
Containers:       cilium             Running: 1
                  cilium-operator    Running: 1
Cluster Pods:     5/5 managed by Cilium
Image versions    cilium             quay.io/cilium/cilium:v1.10.15@sha256:adbb60f9d19d641a5496fcfc4769ebb55c1d4cd7699fef752f7c94754194cf83: 1
                  cilium-operator    quay.io/cilium/operator-generic:v1.10.15@sha256:7f2f7b59b541e4f660d65ed0a21d25ab69e49022172529771dd1134315e95c82: 1
```

8. Perform `cilium connectivity test`, multiple runs. Each time I get different results, here are a two examples.

RUN 1
```
📋 Test Report
❌ 9/30 tests failed (4/137 actions), 0 tests skipped, 0 scenarios skipped:
Test [no-policies]:
  ❌ no-policies/client-to-client/ping-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/client-6685f54484-qrnnf (10.0.0.65:0)
Test [allow-all-except-world]:
Test [cluster-entity]:
  ❌ cluster-entity/pod-to-pod/curl-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
Test [client-egress]:
Test [to-entities-world]:
Test [client-ingress-from-other-client-icmp-deny]:
  ❌ client-ingress-from-other-client-icmp-deny/client-to-client/ping-0: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99:0)
Test [client-egress-to-echo-expression-deny]:
Test [client-egress-l7]:
Test [to-fqdns]:
  ❌ to-fqdns/pod-to-world/https-to-one-one-one-one-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> one-one-one-one-https (one.one.one.one:443)
connectivity test failed: 9 tests failed
```

RUN 2
```
📋 Test Report
❌ 12/30 tests failed (11/139 actions), 0 tests skipped, 0 scenarios skipped:
Test [no-policies]:
  ❌ no-policies/pod-to-service/curl-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node (echo-same-node:8080)
  ❌ no-policies/pod-to-hostport/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (192.168.10.63:40000)
Test [all-ingress-deny]:
Test [cluster-entity]:
  ❌ cluster-entity/pod-to-pod/curl-0: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
  ❌ cluster-entity/pod-to-pod/curl-1: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
Test [echo-ingress]:
Test [client-egress]:
  ❌ client-egress/pod-to-pod/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
Test [client-egress-expression]:
  ❌ client-egress-expression/pod-to-pod/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
Test [client-egress-to-echo-service-account]:
  ❌ client-egress-to-echo-service-account/pod-to-pod/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
Test [echo-ingress-l7]:
  ❌ echo-ingress-l7/pod-to-pod-with-endpoints/curl-0-private: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> curl-0-private (10.0.0.120:8080)
Test [echo-ingress-l7-named-port]:
  ❌ echo-ingress-l7-named-port/pod-to-pod-with-endpoints/curl-0-private: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> curl-0-private (10.0.0.120:8080)
Test [echo-ingress-from-other-client-deny]:
Test [client-egress-l7-method]:
Test [dns-only]:
  ❌ dns-only/pod-to-pod/curl-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
  ❌ dns-only/pod-to-pod/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)
connectivity test failed: 12 tests failed
```

RUN 3
```
📋 Test Report
❌ 11/30 tests failed (8/119 actions), 0 tests skipped, 0 scenarios skipped:
Test [no-policies]:
  ❌ no-policies/pod-to-local-nodeport/curl-1: cilium-test/client-6685f54484-2g7zs (10.0.0.250) -> cilium-test/echo-same-node (echo-same-node:8080)
Test [all-ingress-deny]:
Test [echo-ingress]:
  ❌ echo-ingress/pod-to-pod/curl-1: cilium-test/client-6685f54484-2g7zs (10.0.0.250) -> cilium-test/echo-same-node-658b799748-wq4xz (10.0.0.163:8080)
Test [client-ingress-from-other-client-icmp-deny]:
Test [client-ingress-to-echo-named-port-deny]:
Test [client-egress-to-cidr-deny]:
Test [client-egress-l7-method]:
  ❌ client-egress-l7-method/pod-to-pod-with-endpoints/curl-0-public: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> curl-0-public (10.0.0.163:8080)
  ❌ client-egress-l7-method/pod-to-pod-with-endpoints/curl-0-private: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> curl-0-private (10.0.0.163:8080)
  ❌ client-egress-l7-method/pod-to-pod-with-endpoints/curl-0-privatewith-header: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> curl-0-privatewith-header (10.0.0.163:8080)
Test [client-egress-l7]:
  ❌ client-egress-l7/pod-to-pod/curl-0: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> cilium-test/echo-same-node-658b799748-wq4xz (10.0.0.163:8080)
Test [client-egress-l7-named-port]:
Test [dns-only]:
  ❌ dns-only/pod-to-world/http-to-one-one-one-one-0: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> one-one-one-one-http (one.one.one.one:80)
  ❌ dns-only/pod-to-pod/curl-0: cilium-test/client2-5d4c998f5d-jbr2k (10.0.0.245) -> cilium-test/echo-same-node-658b799748-wq4xz (10.0.0.163:8080)
Test [to-fqdns]:
connectivity test failed: 11 tests failed
```

In the "Revelevant log output" I will upload the complete output of the first run of `cilium connectivity test`.

When looking at the output of some tests I can see errors talking about: `failed: error dialing backend: EOF`. Is anyone experiencing the same issues and knows how to solve this?

### Cilium Version

```
root@test:/home/sander# cilium version
cilium-cli: v0.12.4 compiled with go1.19.1 on linux/amd64
cilium image (default): v1.12.2
cilium image (stable): v1.12.2
cilium image (running): v1.12.2
```

### Kernel Version

```
root@test:/home/sander# uname -a
Linux test2 5.15.0-48-generic #54-Ubuntu SMP Fri Aug 26 13:26:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
root@test:/home/sander# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.4+k3s1", GitCommit:"c3f830e9b9ed8a4d9d0e2aa663b4591b923a296e", GitTreeState:"clean", BuildDate:"2022-08-25T03:45:26Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.4+k3s1", GitCommit:"c3f830e9b9ed8a4d9d0e2aa663b4591b923a296e", GitTreeState:"clean", BuildDate:"2022-08-25T03:45:26Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
```

### Sysdump

```
root@test:/home/sander# cilium sysdump
🔍 Collecting sysdump with cilium-cli version: v0.12.4, args: [sysdump]
🔍 Collecting Kubernetes nodes
🔍 Collect Kubernetes nodes
🔍 Collecting Kubernetes events
🔍 Collect Kubernetes version
🔍 Collecting Kubernetes pods
🔍 Collecting Kubernetes namespaces
🔍 Collecting Kubernetes services
🔍 Collecting Kubernetes pods summary
🔍 Collecting Cilium network policies
🔍 Collecting Kubernetes network policies
🔍 Collecting Kubernetes endpoints
🔍 Collecting Cilium egress NAT policies
🔍 Collecting Cilium cluster-wide network policies
🔍 Collecting Cilium endpoints
🔍 Collecting Cilium local redirect policies
🔍 Collecting Cilium nodes
🔍 Collecting Cilium identities
🔍 Collecting CiliumClusterwideEnvoyConfigs
🔍 Collecting Ingresses
🔍 Collecting Cilium etcd secret
🔍 Collecting CiliumEnvoyConfigs
🔍 Collecting the Cilium configuration
🔍 Collecting the Cilium daemonset(s)
🔍 Collecting the Hubble Relay configuration
🔍 Collecting the Hubble daemonset
🔍 Collecting the Hubble Relay deployment
🔍 Collecting the 'clustermesh-apiserver' deployment
🔍 Collecting the CNI configuration files from Cilium pods
🔍 Collecting the CNI configmap
🔍 Collecting the Hubble UI deployment
🔍 Collecting the Cilium operator deployment
🔍 Collecting gops stats from Cilium pods
🔍 Collecting gops stats from Hubble Relay pods
🔍 Collecting bugtool output from Cilium pods
🔍 Collecting logs from Cilium pods
🔍 Collecting logs from Cilium operator pods
🔍 Collecting gops stats from Hubble pods
⚠️ Deployment "hubble-relay" not found in namespace "kube-system" - this is expected if Hubble is not enabled
⚠️ Deployment "hubble-ui" not found in namespace "kube-system" - this is expected if Hubble UI is not enabled
⚠️ Deployment "clustermesh-apiserver" not found in namespace "kube-system" - this is expected if 'clustermesh-apiserver' isn't enabled
🔍 Collecting logs from Hubble Relay pods
🔍 Collecting logs from Hubble UI pods
🔍 Collecting bugtool output from Tetragon pods
🔍 Collecting platform-specific data
🔍 Collecting kvstore data
🔍 Collecting Hubble flows from Cilium pods
🔍 Collecting logs from 'clustermesh-apiserver' pods
🔍 Collecting logs from Hubble pods
⚠️ The following tasks failed, the sysdump may be incomplete:
⚠️ [11] Collecting Cilium egress NAT policies: failed to collect Cilium egress NAT policies: the server could not find the requested resource (get ciliumegressnatpolicies.cilium.io)
⚠️ [12] Collecting Cilium local redirect policies: failed to collect Cilium local redirect policies: the server could not find the requested resource (get ciliumlocalredirectpolicies.cilium.io)
⚠️ [17] Collecting CiliumClusterwideEnvoyConfigs: failed to collect CiliumClusterwideEnvoyConfigs: the server could not find the requested resource (get ciliumclusterwideenvoyconfigs.cilium.io)
⚠️ [18] Collecting CiliumEnvoyConfigs: failed to collect CiliumEnvoyConfigs: the server could not find the requested resource (get ciliumenvoyconfigs.cilium.io)
⚠️ [23] Collecting the Hubble Relay configuration: failed to collect the Hubble Relay configuration: configmaps "hubble-relay-config" not found
⚠️ Please note that depending on your Cilium version and installation options, this may be expected
🗳 Compiling sysdump
✅ The sysdump has been saved to /home/sander/cilium-sysdump-20220922-120501.zip
```

### Relevant log output

```shell
[=] Test [no-policies]
...
  [-] Scenario [no-policies/client-to-client]
  [.] Action [no-policies/client-to-client/ping-0: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99:0)]
  [.] Action [no-policies/client-to-client/ping-1: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/client-6685f54484-qrnnf (10.0.0.65:0)]
  [-] Scenario [no-policies/pod-to-service]
  [.] Action [no-policies/pod-to-service/curl-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node (echo-same-node:8080)]
  ❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --connect-timeout 5 --output /dev/null http://echo-same-node:8080" failed: error dialing backend: EOF
  ℹ️  curl output:

[=] Test [all-ingress-deny]
......
  ℹ️  📜 Applying CiliumNetworkPolicy 'all-ingress-deny' to namespace 'cilium-test'..
  [-] Scenario [all-ingress-deny/pod-to-pod]
  [.] Action [all-ingress-deny/pod-to-pod/curl-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)]
  [.] Action [all-ingress-deny/pod-to-pod/curl-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cilium-test/echo-same-node-658b799748-5skr9 (10.0.0.120:8080)]
  [-] Scenario [all-ingress-deny/pod-to-cidr]
  [.] Action [all-ingress-deny/pod-to-cidr/cloudflare-1001-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cloudflare-1001 (1.0.0.1:80)]
  [.] Action [all-ingress-deny/pod-to-cidr/cloudflare-1001-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cloudflare-1001 (1.0.0.1:80)]
  [.] Action [all-ingress-deny/pod-to-cidr/cloudflare-1111-0: cilium-test/client2-5d4c998f5d-95dvf (10.0.0.99) -> cloudflare-1111 (1.1.1.1:80)]
  [.] Action [all-ingress-deny/pod-to-cidr/cloudflare-1111-1: cilium-test/client-6685f54484-qrnnf (10.0.0.65) -> cloudflare-1111 (1.1.1.1:80)]
  ℹ️  Cilium agent kube-system/cilium-fk424 logs since 2022-09-22 12:15:38.794401542 +0000 UTC m=+11.861543106:
2022-09-22T12:15:38.897518058Z level=info msg="Policy Add Request" ciliumNetworkPolicy="[&{EndpointSelector:{\"matchLabels\":{\"k8s:io.kubernetes.pod.namespace\":\"cilium-test\"}} NodeSelector:{} Ingress:[{IngressCommonRule:{FromEndpoints:[] FromRequires:[] FromCIDR: FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]} ToPorts:[] ICMPs:[]}] IngressDeny:[] Egress:[] EgressDeny:[] Labels:[k8s:io.cilium.k8s.policy.derived-from=CiliumNetworkPolicy k8s:io.cilium.k8s.policy.name=all-ingress-deny k8s:io.cilium.k8s.policy.namespace=cilium-test k8s:io.cilium.k8s.policy.uid=baf4d90b-cc13-49f0-ab83-7b6a5aa2a755] Description:}]" policyAddRequest=64995a68-3add-4119-9f3a-47770432fb9a subsys=daemon
2022-09-22T12:15:38.897536838Z level=info msg="Policy imported via API, recalculating..." policyAddRequest=64995a68-3add-4119-9f3a-47770432fb9a policyRevision=84 subsys=daemon
2022-09-22T12:15:38.897541119Z level=info msg="Imported CiliumNetworkPolicy" ciliumNetworkPolicyName=all-ingress-deny k8sApiVersion= k8sNamespace=cilium-test subsys=k8s-watcher
2022-09-22T12:15:39.102755962Z level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=83 desiredPolicyRevision=84 endpointID=808 identity=6872 ipv4= ipv6= k8sPodName=/ subsys=endpoint
2022-09-22T12:15:39.157175237Z level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=83 desiredPolicyRevision=84 endpointID=745 identity=11055 ipv4= ipv6= k8sPodName=/ subsys=endpoint
2022-09-22T12:15:39.270803353Z level=info msg="Rewrote endpoint BPF program" containerID= datapathPolicyRevision=83 desiredPolicyRevision=84 endpointID=977 identity=15677 ipv4= ipv6= k8sPodName=/ subsys=endpoint

  ❌ Error finalizing 'all-ingress-deny': getting policy revisions for Cilium agents: error dialing backend: EOF
```

### Anything else?

I am aware of systemd version 245 and above to overriding the `rp_filter` setting of cilium network interfaces. But using cilium version 1.12.0 it is configured to prevent this using `99-zzz-override_cilium.conf` in `/etc/sysctl.d` folder.

```
root@test:/etc/sysctl.d# systemd --version
systemd 249 (249.11-0ubuntu3.4)
```

```
root@test: cat /etc/sysctl.d/99-zzz-override_cilium.conf

Disable rp_filter on Cilium interfaces since it may cause mangled packets to be dropped
-net.ipv4.conf.lxc*.rp_filter = 0
-net.ipv4.conf.cilium_*.rp_filter = 0
The kernel uses max(conf.all, conf.{dev}) as its value, so we need to set .all. to 0 as well.
Otherwise it will overrule the device specific settings.
net.ipv4.conf.all.rp_filter = 0
```

```
root@test:/home/sander# sysctl -a | grep rp_filter
net.ipv4.conf.all.arp_filter = 0
net.ipv4.conf.all.rp_filter = 0
net.ipv4.conf.cilium_host.arp_filter = 0
net.ipv4.conf.cilium_host.rp_filter = 0
net.ipv4.conf.cilium_net.arp_filter = 0
net.ipv4.conf.cilium_net.rp_filter = 0
net.ipv4.conf.cilium_vxlan.arp_filter = 0
net.ipv4.conf.cilium_vxlan.rp_filter = 0
net.ipv4.conf.default.arp_filter = 0
net.ipv4.conf.default.rp_filter = 2
net.ipv4.conf.ens160.arp_filter = 0
net.ipv4.conf.ens160.rp_filter = 2
net.ipv4.conf.lo.arp_filter = 0
net.ipv4.conf.lo.rp_filter = 2
net.ipv4.conf.lxc08cdeca53a3c.arp_filter = 0
net.ipv4.conf.lxc08cdeca53a3c.rp_filter = 0
net.ipv4.conf.lxc28f333328998.arp_filter = 0
net.ipv4.conf.lxc28f333328998.rp_filter = 0
net.ipv4.conf.lxc2aee4ee0e2d4.arp_filter = 0
net.ipv4.conf.lxc2aee4ee0e2d4.rp_filter = 0
net.ipv4.conf.lxc664a00f63b63.arp_filter = 0
net.ipv4.conf.lxc664a00f63b63.rp_filter = 0
net.ipv4.conf.lxc_health.arp_filter = 0
net.ipv4.conf.lxc_health.rp_filter = 0
net.ipv4.conf.lxca05f5cf4c171.arp_filter = 0
net.ipv4.conf.lxca05f5cf4c171.rp_filter = 0
```

I have also tried the same steps to reproduce this in Ubuntu 20.04 and I get the same issues.

I have also tried cilium versions 1.11.3, 1.10.15 but keep getting the same results unfortunately. 

I am wondering what I am missing in my configurations. Or are the cilium tests misleading?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Zandschuur** [comment]
Allright. I think I am heading somewhere...

Today I have reproduced the steps above on my Desktop PC using a VirtualBox virtual machine. All cilium tests are succesful!

This means it's something on my Supermicro ESXI server running ESXI 7.0 update 2. Maybe it could be some network adapter settings in my VM. Or it could be my virtual switch.. Still investigating...

**Comment by patrickjahns** [comment]
Running into similar issues currently. The setup we are investigating this issue on, are debian hosts which form an overlay network directly via wireguard.
What I fail to understand is, where the `failed: error dialing backend: EOF` is coming from. I suspect by now, that it is actually a k3s component ( the tunnel server running the connection between the server<->agent which proxies all requests from the cp/server to the agent. And this tunnel for some reason is not always working as expected. Though I fail to dig deeper

**Comment by patrickjahns** [comment]
Short update - I've dug into how cilium connectivity tests are performed. They indeed use `kubectl exec` or other commands. 
Digging into this, I can see that there are issues related to this on the k3s repo - namely https://github.com/k3s-io/k3s/issues/5633

Currently running k3s1.23.10 - which does not have any fixes for this. Maybe you double check the k3s versions or disable the egress selector as mentioned in the issue. Maybe that also works in your environment

**Comment by Zandschuur** [comment]
I will check today and report my findings with the flag `--egress-selector-mode=disabled`. Good job in findings this. In the meantime I was able to get it to work on my ESXI system with k3s version v1.21.14+k3s1 and cilium version 1.9.18.

**Comment by Zandschuur** [comment]
I just checked with the following component versions with my original configurations:

k3s: `v1.24.4+k3s1`
cilium: `v1.12.2`

`cilium connectivity test`

```
ℹ️  Single-node environment detected, enabling single-node connectivity test
ℹ️  Monitor aggregation detected, will skip some flow validation steps
✨ [default] Creating namespace cilium-test for connectivity check...
✨ [default] Deploying echo-same-node service...
✨ [default] Deploying DNS test server configmap...
✨ [default] Deploying same-node deployment...
✨ [default] Deploying client deployment...
✨ [default] Deploying client2 deployment...
⌛ [default] Waiting for deployments [client client2 echo-same-node] to become ready...
⌛ [default] Waiting for deployments [] to become ready...
⌛ [default] Waiting for CiliumEndpoint for pod cilium-test/client2-5d4c998f5d-mxjrk to appear...
⌛ [default] Waiting for CiliumEndpoint for pod cilium-test/client-6685f54484-9k8rl to appear...
⌛ [default] Waiting for pod cilium-test/client2-5d4c998f5d-mxjrk to reach DNS server on cilium-test/echo-same-node-658b799748-k2h5c pod...
⌛ [default] Waiting for pod cilium-test/client-6685f54484-9k8rl to reach DNS server on cilium-test/echo-same-node-658b799748-k2h5c pod...
⌛ [default] Waiting for pod cilium-test/client2-5d4c998f5d-mxjrk to reach default/kubernetes service...
⌛ [default] Waiting for pod cilium-test/client-6685f54484-9k8rl to reach default/kubernetes service...
⌛ [default] Waiting for CiliumEndpoint for pod cilium-test/echo-same-node-658b799748-k2h5c to appear...
⌛ [default] Waiting for Service cilium-test/echo-same-node to become ready...
⌛ [default] Waiting for NodePort 192.168.10.65:30358 (cilium-test/echo-same-node) to become ready...
ℹ️  Skipping IPCache check
🔭 Enabling Hubble telescope...
⚠️  Unable to contact Hubble Relay, disabling Hubble telescope and flow validation: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:4245: connect: connection refused"
ℹ️  Expose Relay locally with:
   cilium hubble enable
   cilium hubble port-forward&
ℹ️  Cilium version: 1.12.2
🏃 Running tests...

[=] Test [no-policies]
......................
[=] Test [allow-all-except-world]
........
[=] Test [client-ingress]
..
[=] Test [all-ingress-deny]
......
[=] Test [all-egress-deny]
........
[=] Test [all-entities-deny]
......
[=] Test [cluster-entity]
..
[=] Test [host-entity]
..
[=] Test [echo-ingress]
..
[=] Test [client-ingress-icmp]
..
[=] Test [client-egress]
..
[=] Test [client-egress-expression]
..
[=] Test [client-egress-to-echo-service-account]
..
[=] Test [to-entities-world]
......
[=] Test [to-cidr-1111]
....
[=] Test [echo-ingress-l7]
......
[=] Test [echo-ingress-l7-named-port]
......
[=] Test [echo-ingress-from-other-client-deny]
....
[=] Test [client-ingress-from-other-client-icmp-deny]
....
[=] Test [client-egress-to-echo-deny]
....
[=] Test [client-ingress-to-echo-named-port-deny]
..
[=] Test [client-egress-to-echo-expression-deny]
..
[=] Test [client-egress-to-echo-service-account-deny]
..
[=] Test [client-egress-to-cidr-deny]
....
[=] Test [health]
.
[=] Test [client-egress-l7-method]
......
[=] Test [client-egress-l7]
........
[=] Test [client-egress-l7-named-port]
........
[=] Test [dns-only]
........
[=] Test [to-fqdns]
........
✅ All 30 tests (149 actions) successful, 0 tests skipped, 0 scenarios skipped.
```

**Conclusion**

Using `--egress-selector-mode=disabled` solves my connection problems resulting in `failed: error dialing backend: EOF`!

**Comment by LarsBingBong** [comment]
Just for info. I've found that on v1.24.4+k3s1 the Cilium connectivity test do not work because of a runc issue > https://github.com/opencontainers/runc/pull/3554 ... this can be fixed with the combo of being on at least v1.24.6+k3s1 and using the `--egress-selector-mode=disabled` parameter.
